pkg/receive/multitsdb.go

```diff
@@ -446,7 +448,7 @@ func (t *MultiTSDB) startTSDB(logger log.Logger, tenantID string, tenant *tenant
 	s, err := tsdb.Open(
 		dataDir,
 		logger,
-		&UnRegisterer{Registerer: reg},
+		reg,
```

Comment:
Issue was happening because we did not have an `UnRegisterer` [here](https://github.com/thanos-io/thanos/pull/5655/files#diff-61eda56d07cd51eb9fde89c6bd9f1fe7e260edb05eb8fb4c1c81aa258f034331R465)